src/Components/Blazor/Blazor/src/Rendering/RendererRegistry.cs

```diff
@@ -15,23 +16,39 @@ internal static class RendererRegistry
 
         private static int _nextId;
         private static Dictionary<int, WebAssemblyRenderer> _renderers = new Dictionary<int, WebAssemblyRenderer>();
+        private readonly static bool _isWebAssembly = RuntimeInformation.IsOSPlatform(OSPlatform.Create("WEBASSEMBLY"));
 
         internal static WebAssemblyRenderer Find(int rendererId)
         {
+            if (!_isWebAssembly)
+            {
+                throw new InvalidOperationException("Renderer registry can only be modified from WebAssembly runtime.");
+            }
+
             return _renderers.ContainsKey(rendererId)
                 ? _renderers[rendererId]
                 : throw new ArgumentException($"There is no renderer with ID {rendererId}.");
         }
 
         public static int Add(WebAssemblyRenderer renderer)
         {
+            if (!_isWebAssembly)
+            {
+                throw new InvalidOperationException("Renderer registry can only be modified from WebAssembly runtime.");
+            }
+
             var id = _nextId++;
             _renderers.Add(id, renderer);
             return id;
         }
 
         public static bool TryRemove(int rendererId)
         {
+            if (!_isWebAssembly)
+            {
+                throw new InvalidOperationException("Renderer registry can only be modified from WebAssembly runtime.");
+            }
+
```

Comment:
Just a thought, which I don't mind whether you do anything with or not :) Instead of having multiple checks for `isWebAssembly` that run each time there are calls, an alternative would be changing the static constructor so that the static `_renderers` dictionary doesn't even get instantiated in non-wasm cases.

Then there would be no need to check `isWebAssembly` in each of these methods - they would just nullref if they were called on the wrong platform (and they never should be called like this - it could only happen if we introduced a bug since it's `internal` anyway - so a nullref is a reasonable level of error handling).

Reply:
Yep, I can make this change.